### ISSUE TITLE
server resources fail to start after node failure

### ISSUE THREAD

**Issue by fabbione** [ClusterLabs/anvil#329]
Usual CI setup here, complete the deployment and you get 2 servers (I had 3 manually configured).

kill the node where the servers are running (node01 in our scenario), and resources fail to start on node02:

  * an-test-deploy1     (ocf::alteeve:server):   FAILED an-a01n02 (blocked)
  * an-test-deploy2     (ocf::alteeve:server):   FAILED an-a01n02 (blocked)
  * an-test-deploy3     (ocf::alteeve:server):   FAILED an-a01n02 (blocked)

Failed Resource Actions:
  * an-test-deploy2_start_0 on an-a01n02 'not configured' (6): call=76, status='complete', last-rc-change='Sat May  6 08:29:22 2023', queued=0ms, exec=31534ms
  * an-test-deploy3_start_0 on an-a01n02 'not configured' (6): call=75, status='complete', last-rc-change='Sat May  6 08:29:22 2023', queued=0ms, exec=31625ms
  * an-test-deploy1_start_0 on an-a01n02 'not configured' (6): call=74, status='complete', last-rc-change='Sat May  6 08:29:22 2023', queued=0ms, exec=31611ms

later on, node01 will rejoin the cluster. fence node02 and:

  * an-test-deploy1     (ocf::alteeve:server):   FAILED an-a01n01 (blocked)
  * an-test-deploy2     (ocf::alteeve:server):   FAILED an-a01n01 (blocked)
  * an-test-deploy3     (ocf::alteeve:server):   FAILED an-a01n01 (blocked)

Failed Resource Actions:
  * an-test-deploy2_start_0 on an-a01n01 'error' (1): call=79, status='Timed Out', exitreason='Resource agent did not complete within 60s', last-rc-change='Sat May  6 08:32:40 2023', queued=0ms, exec=60010ms
  * an-test-deploy3_start_0 on an-a01n01 'error' (1): call=77, status='Timed Out', exitreason='Resource agent did not complete within 60s', last-rc-change='Sat May  6 08:32:40 2023', queued=0ms, exec=60011ms
  * an-test-deploy1_start_0 on an-a01n01 'error' (1): call=80, status='Timed Out', exitreason='Resource agent did not complete within 60s', last-rc-change='Sat May  6 08:32:40 2023', queued=0ms, exec=60008ms

waited several minutes for drbd to resync status, scancore to do it´s magic, but resources are not being recovered.

**Comment by fabbione** [comment]
had to kick them back online with a pcs resource cleanup.

**Comment by digimer** [comment]
reproduced

**Comment by fabbione** [comment]
Fix is not complete.

deploy with CI, servers will start on node01 (usually).

kill node01

watch the servers start on node02 -> now this is fixed and works.

node01 rejoins the cluster and things go KABOOM.

For quite sometime you see:
[root@an-a01n01 ~]# drbdadm status
an-test-deploy1 role:Secondary
  disk:Consistent
  an-a01n02 connection:StandAlone

an-test-deploy2 role:Secondary
  disk:Consistent
  an-a01n02 connection:StandAlone

then:
[root@an-a01n01 ~]# drbdadm status
an-test-deploy1 role:Secondary
  disk:Consistent
  an-a01n02 connection:Connecting

an-test-deploy2 role:Secondary
  disk:Consistent
  an-a01n02 connection:Connecting

No changes from node02.

[root@an-a01n02 ~]#  drbdadm status
an-test-deploy1 role:Primary
  disk:UpToDate
  an-a01n01 connection:StandAlone

an-test-deploy2 role:Primary
  disk:UpToDate
  an-a01n01 connection:StandAlone

when node01 rejoins the pacemaker cluster, it will try to migrate the servers to node01 for no reasons. They are running fine on node02.

Migration fails:
  * an-test-deploy1     (ocf::alteeve:server):   FAILED (blocked) [ an-a01n01 an-a01n02 ]
  * an-test-deploy2     (ocf::alteeve:server):   Migrating an-a01n02

Failed Resource Actions:
  * an-test-deploy1_migrate_to_0 on an-a01n02 'error' (1): call=113, status='complete', last-rc-change='Wed May 10 06:00:28 2023', queued=0ms, exec=34235ms

There are several stages of failures for each resource, grabbing only the last one.

**Comment by digimer** [comment]
I don't have a great reproducer yet, but I have got it to trigger a couple times now. I've fixed the issue already where the restored node triggers a migration back, but the FAILED state still occurs, albeit less often.

**Comment by digimer** [comment]
Found the issue I think, working on a solution now

**Comment by digimer** [comment]
I've not been able to reproduce this since these changes were added. Please re-open if the issue returns.

**Comment by fabbione** [comment]
bug is still present and reproducible each time with current CI packages.

# rpm -qa|grep anvil
anvil-core-2.90-1.2138.133d.el8.noarch
anvil-node-2.90-1.2138.133d.el8.noarch

Kill node01 -> see 2 servers migrate to node02 -> when node01 rejoins, things go south.

node01 drbd is ok, node02 drbd is in standalone mode on both nodes, pacemaker attempts to migrate the server and fails.

attempting to do a pcs resource clean unblock drbd on node01, but not on node02.

**Comment by fabbione** [comment]
This is still an issue with PR#335. The trigger is pcmk starting to migrate a resource before the DRBD resource underneath is in a state that can migrate.

**Comment by fabbione** [comment]
Issue is only partially fixed in latest anvil.

node02 was running all VMs. node02 failed, resources started correctly on node01, when node02 rejoined:

```
  * delay_node2 (stonith:fence_delay):   Started an-a01n02
  * an-test-deploy1     (ocf::alteeve:server):   Started an-a01n01
  * an-test-deploy2     (ocf::alteeve:server):   Started an-a01n01
  * an-test-deploy3     (ocf::alteeve:server):   Started an-a01n01
  * an-test-deploy4     (ocf::alteeve:server):   FAILED (blocked) [ an-a01n02 an-a01n01 ]
  * an-test-deploy5     (ocf::alteeve:server):   Started an-a01n01

Failed Resource Actions:
  * an-test-deploy4_migrate_to_0 on an-a01n01 'error' (1): call=113, status='complete', last-rc-change='Sat Jun 17 07:25:45 2023', queued=0ms, exec=33071ms

```

test-deploy4 had different attempts of being migrated to node02.

drbd was originally not connected, later in the process it was restored correctly to sync.

test-deploy4 is still responding to ping and ssh, so the node is alive.

over time, the resource will eventually recover in pacemaker:

```
  * delay_node2 (stonith:fence_delay):   Started an-a01n02
  * an-test-deploy1     (ocf::alteeve:server):   Started an-a01n01
  * an-test-deploy2     (ocf::alteeve:server):   Started an-a01n01
  * an-test-deploy3     (ocf::alteeve:server):   Started an-a01n01
  * an-test-deploy4     (ocf::alteeve:server):   Started an-a01n02
  * an-test-deploy5     (ocf::alteeve:server):   Started an-a01n01
```

**Comment by fabbione** [comment]
Reducing severity to "High" since the resources are all recovered, but it´s not the intended design behavior.

**Comment by fabbione** [comment]
The issue exists also when creating multiple nodes in a strict serial sequence:

```
  * an-test-deploy1     (ocf::alteeve:server):   Started an-a01n01
  * an-test-deploy2     (ocf::alteeve:server):   Started an-a01n01
  * an-test-deploy3     (ocf::alteeve:server):   Started an-a01n01
  * an-test-deploy4     (ocf::alteeve:server):   Started an-a01n02
  * an-test-deploy5     (ocf::alteeve:server):   Started an-a01n01
```

**Comment by fabbione** [comment]
to keep things clean, resources are recovered. Let´s file another issue specific to location.
